### PR TITLE
Feature/sku-selector-editor

### PR DESCRIPTION
**Pull request by bellacorbo**
This PR implements the Editor for the Sku Selector

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,10 +27,13 @@
...
     "raw-loader": "^0.5.1",
     "react": "^0.14.0",
+    "react-dnd": "^2.0.2",
+    "react-dnd-html5-backend": "^2.0.0",
     "react-proxy-loader": "^0.3.4",
     "react-transform-catch-errors": "^1.0.0",
     "react-transform-hmr": "^1.0.0",
     "redbox-react": "^1.0.1",
+    "sortablejs": "^1.4.2",
     "style-loader": "^0.9.0",
```
Comment (tamorim) [inline package.json]: @bellacorbo These drag and drop components are being used?

File: src/components/AddToCartButton/AddToCartButton.less [hunk 2]
```diff
@@ -1,4 +1,5 @@
-@import '../../utils/base';
+@import "~vtex-css-utils/less/_css-utils";
+@import "~utils/base/base";
```
Comment (tamorim) [inline src/components/AddToCartButton/AddToCartButton.less]: @bellacorbo @brenoc She should remove the vtex-css-utils ref here now, right?

File: src/components/SkuSelector/SelectVariation/VariationButton/style.less [hunk 3]
```diff
@@ -0,0 +1,68 @@
+@import "~vtex-css-utils/less/_css-utils";
+@import "~utils/base/base";
+
```
Comment (tamorim) [inline src/components/SkuSelector/SelectVariation/VariationButton/style.less]: @brenoc @bellacorbo Remove vtex-css-utils ref here too, if the CategoryPage merges first...

File: src/components/SkuSelector/style.less [hunk 4]
```diff
@@ -1,4 +1,5 @@
-@import '../../utils/base';
+@import "~vtex-css-utils/less/_css-utils";
+@import "~utils/base/base";
```
Comment (tamorim) [inline src/components/SkuSelector/style.less]: @bellacorbo @brenoc and here

File: src/components/AddToCartButton/AddToCartButton.less (hunk 2 above)
Comment (brenoc) [inline src/components/AddToCartButton/AddToCartButton.less]: I'll merge category page, then @bellacorbo should rebase this

File: src/editors/AddToCartButtonEditor/AddToCartButtonEditor.less [hunk 5]
```diff
@@ -1,4 +1,5 @@
-@import "../../utils/base";
+@import "~utils/base";
+@import "~vtex-css-utils/less/_css-utils";
```
Comment (tamorim) [inline src/editors/AddToCartButtonEditor/AddToCartButtonEditor.less]: @bellacorbo @brenoc here too

File: src/editors/SkuSelectorEditor/SelectVariationEditor/VariationButtonEditor/style.less [hunk 6]
```diff
@@ -0,0 +1,38 @@
+@import "~vtex-css-utils/less/_css-utils";
+@import "~utils/base/base";
+
```
Comment (tamorim) [inline src/editors/SkuSelectorEditor/SelectVariationEditor/VariationButtonEditor/style.less]: @bellacorbo @brenoc Just to help out!

File: src/editors/SkuSelectorEditor/SelectVariationEditor/style.less [hunk 7]
```diff
@@ -0,0 +1,98 @@
+@import "~vtex-css-utils/less/_css-utils";
+@import "~utils/base/base";
+
```
Comment (tamorim) [inline src/editors/SkuSelectorEditor/SelectVariationEditor/style.less]: @bellacorbo @brenoc outlining, sorry :)

File: src/editors/SkuSelectorEditor/style.less [hunk 8]
```diff
@@ -0,0 +1,42 @@
+@import "~vtex-css-utils/less/_css-utils";
+@import "~utils/base/base";
+
```
Comment (tamorim) [inline src/editors/SkuSelectorEditor/style.less]: @bellacorbo @brenoc :)

File: src/utils/base/base.less [hunk 9]
```diff
@@ -0,0 +1,2 @@
+@import "theme-variables";
+@import "theme";
```
Comment (tamorim) [inline src/utils/base/base.less]: @bellacorbo @brenoc this structure will conflict with the actual one, where we uses base files on the utils folder directly

**Comment by tamorim** [comment]
@brenoc @bellacorbo @amoreira Updated with the minimal changes to use the `sku-selector` app!

**Comment by brenoc** [comment]
Finished reviewing, it's OK.

@bellacorbo we need to revisit SkuSelector to be Plug&Play. The way that it is now the app that consumes sku-selector needs to write a bunch of code ([all of this](https://github.com/vtex-apps/theme/blob/feature/sku-selector-editor/src/components/Product/Product.js#L11-L85))

Ideally `Product.js` (or any other app that wants to use sku-selector app) would just need to write this inside render:

```js
<Area id="product/sku-selector" skus={skus}/>
```

**Comment by bellacorbo** [comment]
@breno done in f391f5933ce65125fbda18bbbfae827b3226b733 and in https://github.com/vtex-apps/sku-selector/commit/09f75f3b93e405ce575cbdfbeb51d109ac8dbd53

I had to pass a callback as prop to the sku-selector because of the cartButton interaction with the selected sku, but besides that, the structure of the render is as you showed above. :)

**Comment by dfosco** [comment]
@tamorim @amoreira @brenoc I rebased the branch, but broke something along the way :sweat_smile:

Thor, take another look whenever you can — but I agree with Anderson that we should merge it ASAP. Too much work has been done here already, I doubt that there are issues big enough to stop it from going into the MVP. If there are, we'll deal with them later.

**Pull request merged**